src/components/TodoModal/TodoModal.tsx

```diff
@@ -3,40 +1,85 @@
+import { useAppDispatch, useAppSelector } from '../../app/hook';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+export const TodoModal: FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  const dispatch = useAppDispatch();
+  const todo = useAppSelector(state => state.currentTodo);
+
+  const handleCloseModal = () => {
+    dispatch(currentTodoSlice.actions.setCurrentTodo(null));
+  };
+
+  useEffect(() => {
+    if (todo) {
+      setLoading(true);
+
+      getUser(todo.userId)
+        .then(setUser)
+        .finally(() => setLoading(false));
+    }
+  }, [todo]);
+
+  if (!todo) {
+    return null;
+  }
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {loading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{todo.id}
+            </div>
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={handleCloseModal}
+            />
+          </header>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {todo.title}
+            </p>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+            <p className="block" data-cy="modal-user">
+              <strong
+                className={cn({
+                  'has-text-danger': !todo.completed,
+                  'has-text-success': todo.completed,
+                })}
+              >
+                {todo.completed ? `Done` : `Planned`}
+              </strong>
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+              {' by '}
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+              <a href={`mailto:${user?.email}`}>
+                {user?.name}
+              </a>
```

Comment:
Ensure that `user` is not null before rendering the email link to avoid potential runtime errors. You might want to conditionally render this part only if `user` is available.